### DISCUSSION TITLE
Created Module runs from the local file system, but not when loaded remotely.

### DISCUSSION THREAD

**Discussion by anoduck** [gethinode/hinode#1330] [Help] [answered]
The module is [timelinejs](https://github.com/anoduck/mod-timelinejs), and I have been working on it for a while. To help troubleshoot all the issues I have had with it I created a web site to test it out on, [Hinode Test Site](https://testhinode.netlify.app/].

All changes have been pushed to the repository, the repository reflects those changes, all branches are up to date, and are even with each other.

Regardless, if the module is loaded from the local file system by the use of a [replacements statement](https://github.com/anoduck/hinode-testsite/blob/6be6ea3c5d81bdefd87e735d1417241192822db7/config/_default/hugo.toml#L88), `replacements = "github.com/anoduck/mod-timelinejs -> ../../mod-timelinejs"`, the page that uses the module is rendered correctly and everything is as it should, but if the replacements statement is comment out, which requires the module to be loaded from the repository, then the module is not bundled along with other resources and the content is not rendered. 

So, If both the local copy of the repository and the remote copy of the repository are exactly identical, then why does the module fail to be bundled when the remote repository is used?

**Comment by anoduck** [comment]
@d-oit Dom, I know you are busy and have your own projects, but could you take a look at this and tell me why it is not working on production sites?

**Reply by d-oit** [reply, marked as answer]
Try to remove the dist/ folder from the .gitignore or mount the package like this:

https://discourse.gohugo.io/t/esbuild-looks-like-we-can-finally-get-solid-hugo-modules-support/28757

If it works local then you have a dist folder local but not in your github repo to mount for the build. You could check all download files in the vendor folder of your hinode site.
Your github repo must include a full working build version.  

[Use the mod-leaf](https://github.com/gethinode/mod-leaflet/blob/main/config.toml) as reference 
Otherwise, you load the js / css on all pages.

**Reply by anoduck** [reply]
@d-oit I didn't know the `.gitignore` template for Emacs included `./dist`, no wonder it wasn't working. That is one lesson I am going to feel for a while. (－‸ლ)

```text
  .-'---`-.
,'          `.
|             \
|              \
\           _  \
,\  _    ,'-,/-)\
( * \ \,' ,' ,'-)
 `._,)     -',-')
   \/         ''/
    )        / /
   /       ,'-'
```
Time to rebuild my site, and get it online. 

Sometimes you need another pair of eyes looking at things.
Thanks, Dom

**Reply by d-oit** [reply]
ok, tested out: it works if you remove the timelinejs -> js in the module name to something else. Otherwise, the Hinode bundle is not working 😕


params.toml 
[debugging]
    showJS = true
    
WARN  Processing pattern for 'js/timeline.js': {js/modules/timeline/**.js}
WARN  - Processing file: /js/modules/timeline/autoload.js
WARN  - Processing file: /js/modules/timeline/timeline.js


A few other problems to fix - like csp or use the data folder for the json

<img width="953" alt="image" src="https://github.com/user-attachments/assets/bc6379bb-7213-44b9-9e07-0883d74d990e" />



https://gohugo.io/content-management/data-sources/

**Reply by d-oit** [reply]
https://github.com/d-oit/mod-timelinejs-main
@anoduck my test repo

**Reply by anoduck** [reply]
@d-oit Just making sure I understand, your suggesting I take on a different naming scheme for the module? Instead of "timelinejs", something like "knightlabs-timeline"?

**Reply by d-oit** [reply]
Yes, check my repo link.
 It worked with "timeline" as the module name but with a few security errors in the browser console.

Happy new year 😉

**Reply by anoduck** [reply]
@d-oit Boy... I am so grateful for your help. You hung the moon as far as I am concerned. 

Corrected loading of the sass stylesheet, as well as many other significant improvements. So your timeline should be formatted now. As for the minor security issues, my understanding of them was they originated from Knight-Labs TimelineJS itself and is related to the numerous resources TimelineJS is allowed to pull from. I might be wrong about this, though.

**Reply by d-oit** [reply]
Maybe helpful for the next module: https://github.com/d-oit/streamlit-generator-hinode-mod

**Reply by anoduck** [reply]
Now, that's just cheating... 😆 😝 😆
